File: old/session12Strings/BadLCS.cc
#include <string>

using namespace std;

int lcs(string A, string B)
{
	// must be static variable for dynamic programming!
	// static int *memo= new int[m*n];
	if (A.length() == 0 || B.length() == 0)
		return 0;        //O(1)
	
	if (A[0] == B[0])
		return 1 + lcs(A.substr(1), b.substr(1));    //O(min(len(A),len(B))
  return max(lcs(A.substr(1),B), lcs(A,B.substr(1)));    //O(2^n)
}

File: old/session12Strings/BadLCS.java
public class BadLCS {
	public int lcs(String a, String b) {
		// dynamic programming: static int memo[][] = new int[m][n];
		if (a.length() == 0 || b.length() == 0)
		return 0;        //O(1)
	
		if (a.charAt(0) == b.charAt(0))
			return 1 + lcs(a.substr(1), b.substr(1));    //O(min(len(A),len(B))
		return max(lcs(a.substr(1),b), lcs(a,b.substr(1)));    //O(2^n)
	}
}
